src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -797,29 +804,32 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     // but using the iterator instead.
 
     const iteratorFn = getIteratorFn(newChildrenIterable);
-    if (typeof iteratorFn !== 'function') {
-      throw new Error('An object is not an iterable.');
-    }
+    invariant(
+      typeof iteratorFn === 'function',
+      'An object is not an iterable. (%s)',
```

Comment:
I think this is correct because if it's not an iterator then we shouldn't have reached this code path.